app/controllers/api/v1/events_controller.rb

```diff
@@ -0,0 +1,69 @@
+# :reek:InstanceVariableAssumption
+# :reek:NilCheck
+
+class Api::V1::EventsController < ApplicationController
+  before_action :authenticate_user!, only: %i[create update destroy]
+  before_action :verify_organizer, only: %i[create update destroy]
+  before_action :set_events, only: %i[index]
+  before_action :set_event, only: %i[show update destroy]
+  before_action :set_organization_event, only: %i[update destroy]
+
+  def index
+    render json: @events
+  end
+
+  def show
+    return head :forbidden unless @event.social? || (current_user&.organization&.events&.find_by(id: @event.id))
+
+    render json: @event
+  end
+
+  def create
+    @event = current_user.events.new(event_params)
+    if @event.save
+      render json: @event, status: :created
+    else
+      render json: @event.errors.full_messages, status: :unprocessable_entity
+    end
+  end
+
+  def update
+    if @event.update(event_params)
+      render json: @event
+    else
+      render json: @event.errors.full_messages, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @event.attendances.destroy_all
+    @event.destroy
+    head :no_content
+  end
+
+  private
+
+  def set_events
+    @events = Event.social
+    @events += current_user.organization.events.confidential if current_user&.organizer?
+  end
+
+  def set_event
+    @event = Event.find_by(id: params[:id])
+    return head :not_found unless @event
+  end
+
+  def set_organization_event
+    return head :forbidden unless current_user.organization.events.find_by(id: @event.id)
```

Comment:
organization_event is not set in this method

Reply:
I'd rename the method, but logic is fine.